packages/opentelemetry-instrumentation/src/instrumentation.ts

```diff
@@ -29,9 +29,9 @@ import * as types from './types';
 /**
  * Base abstract internal class for instrumenting node and web plugins
  */
-export abstract class InstrumentationAbstract<T = any>
+export abstract class InstrumentationAbstract<InstrumentedModuleType = any, Config = types.InstrumentationConfig>
```

Comment:
would this work ?
```suggestion
export abstract class InstrumentationAbstract<InstrumentedModuleType = any, Config extends types.InstrumentationConfig>
```

Reply:
If we make Config required we have to also make InstrumentedModuleType optional (required types can't follow optional types). Is that what you're suggesting?

Reply:
I think it should be a combination - optional with default to `types.InstrumentationConfig` but if a concrete type is specified it must `extend types.InstrumentationConfig` (otherwise it would be allowed to use e.g. `number`).